File: 2024/09/p1.test.lua
local describe = require("lib.test")
local p1 = require(".2024.09.p1")

describe("input", function(it)
  it("example 1", function()
    local example = "2333133121414131402"
    local expected = 1928

    local n = p1.solve(example)

    assert(n == expected, string.format("Expected %d, but was %d", expected, n))
  end)

  it("input", function()
    local fp = io.open("./2024/09/assets/input.txt", "r") or error("Input not found")
    local input = fp:read("*a")
    local expected = 6323641412437

    local n = p1.solve(input)

    assert(n == expected, string.format("Expected %d, but was %d", expected, n))
  end)
end)

File: 2024/10/__tests__/p1.test.ts
import assert from "node:assert";
import { readFile } from "node:fs/promises";
import { describe, it } from "node:test";
import { solve } from "../p1.js";
import path from "node:path";

describe("p1", () => {
  it("example 1", () => {
    const example = `0123
1234
8765
9876`;

    assert.strictEqual(solve(example), 1);
  });

  it("example 2", () => {
    const example = `...0...
...1...
...2...
6543456
7.....7
8.....8
9.....9`;

    assert.strictEqual(solve(example), 2);
  });

  it("example 2", () => {
    const example = `..90..9
...1.98
...2..7
6543456
765.987
876....
987....`;

    assert.strictEqual(solve(example), 4);
  });

  it("example 3", () => {
    const example = `10..9..
2...8..
3...7..
4567654
...8..3
...9..2
.....01`;

    assert.strictEqual(solve(example), 3);
  });


  it("example 4", () => {
    const example = `89010123
78121874
87430965
96549874
45678903
32019012
01329801
10456732`;

    assert.strictEqual(solve(example), 36);
  });

  it("input", async () => {
    const filename = path.join(import.meta.dirname, "../assets/input.txt").replace("/dist/", "/");
    const input = await readFile(filename, "utf8");

    assert.strictEqual(solve(input), 709);
  });
});
